commit 7be300ac4b4deca1176e05a695e95209b74ffa4f
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sat Jul 28 20:31:00 2018 +0200

    Add gcc-8 compatibility
    
    * ./libdde_linux26/contrib/include/linux/compiler-gcc8.h: New file.

diff --git a/libdde_linux26/contrib/include/linux/compiler-gcc8.h b/libdde_linux26/contrib/include/linux/compiler-gcc8.h
new file mode 100644
index 000000000..accf892e2
--- /dev/null
+++ b/libdde_linux26/contrib/include/linux/compiler-gcc8.h
@@ -0,0 +1,25 @@
+#ifndef __LINUX_COMPILER_H
+#error "Please don't include <linux/compiler-gcc8.h> directly, include <linux/compiler.h> instead."
+#endif
+
+#define __used			__attribute__((__used__))
+#define __must_check 		__attribute__((warn_unused_result))
+#define __compiler_offsetof(a,b) __builtin_offsetof(a,b)
+#define __always_inline		inline __attribute__((always_inline))
+
+/*
+ * A trick to suppress uninitialized variable warning without generating any
+ * code
+ */
+#define uninitialized_var(x) x = x
+
+/* Mark functions as cold. gcc will assume any path leading to a call
+   to them will be unlikely.  This means a lot of manual unlikely()s
+   are unnecessary now for any paths leading to the usual suspects
+   like BUG(), printk(), panic() etc. [but let's keep them for now for
+   older compilers]
+
+   gcc also has a __attribute__((__hot__)) to move hot functions into
+   a special section, but I don't see any sense in this right now in
+   the kernel context */
+#define __cold			__attribute__((__cold__))
